tasks/main.yml

```diff
@@ -37,20 +38,22 @@
 
 - name: Set NTP server addresses
   purefa_ntp:
-    state: "{{ purefa_ntp_state | default('present')}}"
+    state: "{{ 'present' if purefa_ntp_servers else 'absent' }}"
     ntp_servers: "{{ purefa_ntp_servers }}"
     fa_url: "{{ purefa_url }}"
     api_token: "{{ purefa_api_token }}"
+  when: purefa_ntp_servers is defined
   tags:
   - purefa_ntp
 
 - name: Set DNS server addresses
   purefa_dns:
-    state: "{{ purefa_dns_state | default('present')}}"
-    domain: "{{ purefa_domain }}"
+    state: "{{ 'present' if purefa_dns_servers else 'absent' }}"
```

Comment:
Is the if/else needed? if `purefa_dns_servers` isn't defined, this task won't run anyway, due to the `when` on line 56

Reply:
It is needed because the user might want to define the variable `purefa_dns_servers` as empty and therefore clear out the current values.

Reply:
Ah I see. Makes sense.